### ISSUE TITLE
Update to Rand 0.6

### ISSUE THREAD

**Issue by coltfred** [dalek-cryptography/ed25519-dalek#56]
Rand 0.6 was released today with support for WASM as well as [other features](https://rust-random.github.io/book/update-0.6.html).

**Comment by 0ndorio** [comment]
An update would be great. Especially because in its current version `ed25519-dalek v1.0.0-pre.0` can't be used be used as crate dependency as long as the crate doesn't contain a direct dependency on `rand v0.5`. 

In any case where this dependency doesn't exist or the crate tries to use `rand v0.6` the following error appears. Reason for this compilation error seems to be a compatibility issue between `rand v0.5.5` and `rand v0.6` combined with type differences based on the `std` feature flag. This results in the issue described in https://github.com/rust-random/rand/issues/645.

```rust
error[E0277]: the trait bound `rngs::jitter::TimerError: std::error::Error` is not satisfied                         
   --> /home/rnd/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.5.5/src/rngs/jitter.rs:269:9                 
    |                                                                                                                
269 |         Error::with_cause(ErrorKind::Unavailable,                                                              
    |         ^^^^^^^^^^^^^^^^^ the trait `std::error::Error` is not implemented for `rngs::jitter::TimerError`      
    |                                                                                                                
    = note: required because of the requirements on the impl of `core::convert::From<rngs::jitter::TimerError>` for `alloc::boxed::Box<(dyn std::error::Error + core::marker::Sync + core::marker::Send + 'static)>`
    = note: required because of the requirements on the impl of `core::convert::Into<alloc::boxed::Box<(dyn std::error::Error + core::marker::Sync + core::marker::Send + 'static)>>` for `rngs::jitter::TimerError`
    = note: required by `rand_core::Error::with_cause`                                                               
                                                                                                                     
error: aborting due to previous error                                                                                
                                                                                                                     
For more information about this error, try `rustc --explain E0277`.                                                  
error: Could not compile `rand`.
```
---
Based on a few quick tests this could either be resolved in one of two ways:

1) Update `rand` from `v0.5.5` to `v0.6`
2) The `std` feature flag could also imply the `rand/std` feature.

     Based on a comment inside the `Cargo.toml` file this currently doesn't happen as it would introduce  `fuchsia-zircon` as additional dependency. This feels like some kind of relic or inconsistency as `curve25519-dalek/std` already includes `rand/std` as an indirect dependency and also seems to be the main reason behind this problem.

**Comment by coltfred** [comment]
@0ndorio I have a PR open for this #57 - Since curve25519 moved to rand 0.6 in `1.0.0-pre.1` I was assuming we would do the same for ed25519. If we want to do something else, let's talk about it in the PR.

**Comment by 0ndorio** [comment]
@coltfred Moving to `rand v0.6` sounds reasonable to me. Nothing to discuss there. On a second thought the second solution actually describes another minor issue related to some dependency inconsistency. This shouldn't effect #57 at all.

**Comment by isislovecruft** [comment]
Fixed in #57
